### PR TITLE
feat(helm): add support for labels, annotations and podAnnotations

### PR DESCRIPTION
**Pull request by h4wkmoon**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Add ability to add labels, annotations and podAnnotations to Helm Charts

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/latest/azuredisk-csi-driver/values.yaml [hunk 1]
```diff
@@ -131,6 +133,7 @@ snapshot:
     name: csi-snapshot-controller
     replicas: 2
     podLabels: {}
+    annotations: {}
```
Comment (andyzhangx) [inline charts/latest/azuredisk-csi-driver/values.yaml:136]: could you use `podAnnotations` naming, that would be consistent with existing `podLabels`, thanks.

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml [hunk 2]
```diff
@@ -4,6 +4,13 @@ metadata:
   name: {{ .Values.controller.name }}
   namespace: {{ .Release.Namespace }}
 {{ include "azuredisk.labels" . | indent 2 }}
+{{- with .Values.controller.labels }}
+{{ . | toYaml | indent 4 }}
+{{- end }}
+{{- with .Values.controller.annotations  }}
```
Comment (aslafy-z) [inline charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:10]:
```suggestion
{{- with .Values.controller.annotations }}
```

File: charts/latest/azuredisk-csi-driver/values.yaml (hunk 1 above)
Comment (aslafy-z) [reply charts/latest/azuredisk-csi-driver/values.yaml:137]: I added `labels`, `annotations` and `podAnnotations`

Comment (h4wkmoon) [reply charts/latest/azuredisk-csi-driver/values.yaml:137]: Thank you.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**